### PR TITLE
fix(route/sjtu): gs

### LINKED ISSUE #12742: 上海交通大学研究生院网站正在改版，RSS已失效
### 路由地址

```routes
/sjtu/gs/tzgg/:type?
```

### 完整路由地址

```fullroutes
/sjtu/gs/tzgg/:type?
```

### 相关文档

https://docs.rsshub.app/university.html#shang-hai-jiao-tong-da-xue-yan-jiu-sheng-tong-zhi-gong-gao

### 预期是什么？

正常运行。

### 实际发生了什么？

gs.sjtu.edu.cn/ 已自上周改版，原有获取方式失效。

### 部署

RSSHub 演示 (https://rsshub.app)

### 部署相关信息

_No response_

### 额外信息

```shell
.
```

### 这不是重复的 issue

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by dzx-dzx**
## 该 PR 相关 Issue / Involved Issue

Close #12742 

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/sjtu/gs/work
/sjtu/gs/enroll/58
/sjtu/gs/enroll/59
/sjtu/gs/enroll/60
/sjtu/gs/enroll/61
/sjtu/gs/enroll/62
/sjtu/gs/train
/sjtu/gs/degree
/sjtu/gs/exchange/67
/sjtu/gs/exchange/68
/sjtu/gs/exchange/69
/sjtu/gs/exchange/70
/sjtu/gs/exchange/71
/sjtu/gs/xsjy
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [ ] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/sjtu/gs.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    const items = await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const detailResponse = await got({
+                    method: 'get',
+                    url: item.link,
+                });
+                const content = cheerio.load(detailResponse.data);
+
+                item.description = (new URL(item.link).hostname === 'mp.weixin.qq.com' ? content('#js_content') : content('.page-content')).html();
```
Comment (TonyRL) [inline lib/v2/sjtu/gs.js:43]: There's a special utility to help you parse articles from `mp.weixin.qq.com` [lib/utils/wechat-mp.js](https://github.com/DIYgod/RSSHub/blob/master/lib/utils/wechat-mp.js)

Comment (TonyRL) [inline lib/v2/sjtu/gs.js:43]: There's a special utility to help you parse articles from `mp.weixin.qq.com` [lib/utils/wechat-mp.js](https://github.com/DIYgod/RSSHub/blob/master/lib/utils/wechat-mp.js)

File: lib/v2/sjtu/radar.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    'sjtu.edu.cn': {
+        _name: '上海交通大学',
+        gs: [
+            {
+                title: '研究生通知公告',
+                docs: 'https://docs.rsshub.app/university.html#shang-hai-jiao-tong-da-xue-yan-jiu-sheng-tong-zhi-gong-gao',
+                source: ['*'],
+                target: '/gs/:type/:num?',
```
Comment (TonyRL) [inline lib/v2/sjtu/radar.js:9]:
```suggestion
                source: ['/announcement/:type'],
                target: '/gs/:type/:num?',
```

File: docs/university.md [hunk 3]
```diff
@@ -2590,11 +2590,23 @@ jsjxy.hbut.edu.cn 证书链不全，自建 RSSHub 可设置环境变量 NODE_TLS
 
 ### 研究生通知公告
 
-<Route author="mzr1996" example="/sjtu/gs/tzgg/pyxx" path="/sjtu/gs/tzgg/:type?" :paramsDesc="['默认列举所有通知公告']">
+<Route author="dzx-dzx" example="/sjtu/gs/enroll/59" path="/sjtu/gs/:type/:num?" :paramsDesc="['类别', '细分类别, 仅对`type`为`enroll`或`exchange`有效']">
```
Comment (TonyRL) [inline docs/university.md:2593]: Do not remove the name of previous author(s)

Comment (dzx-dzx) [reply docs/university.md:2593]: But…Wouldn't it be awkward that for every issue created for this route, the previous author will get pinged for the code he has no knowledge of?

Comment (TonyRL) [reply docs/university.md]: Pinging depends on the entries in `maintainer.js`.

Comment (dzx-dzx) [reply docs/university.md]: Oh OK. Thanks! It's now fixed in https://github.com/DIYgod/RSSHub/pull/12758/commits/b236b25f1f3b05e7ef3528c35a7612dc350590ec .

**Pull request merged**